.github/scripts/create-e2e-test-report.ts

```diff
@@ -145,63 +84,125 @@ async function main() {
         if (existingRun) {
-          existingRun.testSuites.push(testSuite);
+          const existingFile = existingRun.testFiles.find(
+            (file) => file.path === testFile.path,
+          );
+          if (existingFile) {
+            existingFile.testSuites.push(testSuite);
+          } else {
+            existingRun.testFiles.push(testFile);
+          }
         } else {
           testRuns.push(testRun);
         }
       }
     }
 
     for (const testRun of testRuns) {
-      const deduped: { [path: string]: TestSuite } = {};
-      for (const suite of testRun.testSuites) {
-        const existing = deduped[suite.path];
-        // If there is a duplicate, we keep the suite with the latest date
-        if (!existing || existing.date < suite.date) {
-          deduped[suite.path] = suite;
+      for (const testFile of testRun.testFiles) {
+        // Group test suites by name
+        const suitesByName: Record<string, TestSuite[]> = {};
+
+        for (const suite of testFile.testSuites) {
+          if (!suitesByName[suite.name]) {
+            suitesByName[suite.name] = [];
+          }
+          suitesByName[suite.name].push(suite);
         }
-      }
 
-      const suites = Object.values(deduped);
+        // Determine the latest test suite by date and nest attempts
+        const attempts: TestSuite[][] = [];
+        for (const suites of Object.values(suitesByName)) {
+          suites.sort((a, b) => b.date.getTime() - a.date.getTime()); // sort newest first
+          const [latest, ...otherAttempts] = suites;
+          latest.attempts = otherAttempts;
+          attempts.push(otherAttempts);
+        }
 
-      const title = `<strong>${testRun.name}</strong>`;
-      console.log(consoleBold(title));
+        // Remove the nested attempts from the top-level list
+        const attemptSet = new Set(attempts.flat().map((s) => s));
+        testFile.testSuites = testFile.testSuites.filter(
+          (suite) => !attemptSet.has(suite),
+        );
 
-      if (suites.length) {
-        const total = suites.reduce(
+        const total = testFile.testSuites.reduce(
           (acc, suite) => ({
             tests: acc.tests + suite.tests,
             passed: acc.passed + suite.passed,
             failed: acc.failed + suite.failed,
             skipped: acc.skipped + suite.skipped,
+            time: acc.time + suite.time,
           }),
-          { tests: 0, passed: 0, failed: 0, skipped: 0 },
+          { tests: 0, passed: 0, failed: 0, skipped: 0, time: 0 },
         );
 
-        core.summary.addRaw(
-          total.failed ? `\n<details open>\n` : `\n<details>\n`,
+        testFile.tests = total.tests;
+        testFile.passed = total.passed;
+        testFile.failed = total.failed;
+        testFile.skipped = total.skipped;
+        testFile.time = total.time;
+      }
+
+      testRun.testFiles.sort((a, b) => a.path.localeCompare(b.path));
+
+      const title = `<strong>${testRun.name}</strong>`;
```

Comment:
I know this is old code, but is `testRun.name` already sanitized for injection into HTML?